src/main/groovy/com/twcable/grabbit/client/servlets/GrabbitTransactionServlet.groovy

```diff
@@ -57,8 +63,19 @@ class GrabbitTransactionServlet extends SlingSafeMethodsServlet {
             String transactionIDString = request.resource.resourceMetadata[TransactionResource.TRANSACTION_ID_KEY]
             // Guard against the transactionID coming back as null or empty
             if(transactionIDString != null && !transactionIDString.empty) {
-                //At this point we know we at least have something as the "transactionID." It may still not be a valid numeric ID
-                transactionID = transactionIDString.toLong()
+                if(ALL_TRANSACTION_ID.equals(transactionIDString)){
+                    final JobExplorer jobExplorer = configurableApplicationContext.getBean("clientJobExplorer", JobExplorer)
+                    final Collection<String> transactionIds = ClientJobStatus.getAllTransactions(jobExplorer)
+                    response.setStatus(SC_OK)
+                    response.setContentType("text/html")
+                    response.writer.write this.class.getResourceAsStream("GrabbitTransactionResource-all.txt").text + transactionIds.collect {
+                        "<a href=${it}>${it}</a><br/>"
```

Comment:
Hi @jbornemann 
by organizing flow-of-control and forwarding to JSP, i think you mean adding a separate function or TransactionService to handle all the logic and then doing response.sendRedirect to particular JSPs based on is transaction valid, if it's 'all' etc.
But, then again, isn't these changes more than the scope of what's required in this ticket?